.github/workflows/publish-docker.yml

```diff
@@ -0,0 +1,50 @@
+name: Publish Docker Image
```

Comment:
This is an attempt to try to automate some of the infrastructure, automatically publishing new opencode docker images to docker hub whenever new opencode releases are released.